envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/RetryPolicyTest.kt

```diff
@@ -2,34 +2,47 @@ package pl.allegro.tech.servicemesh.envoycontrol
 
 import okhttp3.Headers
 import org.assertj.core.api.Assertions.assertThat
-import org.junit.jupiter.api.BeforeAll
 import org.junit.jupiter.api.Test
-import pl.allegro.tech.servicemesh.envoycontrol.config.EnvoyControlRunnerTestApp
-import pl.allegro.tech.servicemesh.envoycontrol.config.EnvoyControlTestConfiguration
-
-internal class RetryPolicyTest : EnvoyControlTestConfiguration() {
+import org.junit.jupiter.api.extension.RegisterExtension
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.untilAsserted
+import pl.allegro.tech.servicemesh.envoycontrol.config.EnvoyControlExtension
+import pl.allegro.tech.servicemesh.envoycontrol.config.consul.ConsulExtension
+import pl.allegro.tech.servicemesh.envoycontrol.config.containers.EchoServiceExtension
+import pl.allegro.tech.servicemesh.envoycontrol.config.envoy.EnvoyExtension
+
+class RetryPolicyTest {
     companion object {
 
-        private val properties = mapOf(
-            "envoy-control.envoy.snapshot.local-service.retry-policy.per-http-method.GET.enabled" to true,
-            "envoy-control.envoy.snapshot.local-service.retry-policy.per-http-method.GET.retry-on" to listOf("connect-failure", "reset"),
-            "envoy-control.envoy.snapshot.local-service.retry-policy.per-http-method.GET.num-retries" to 3
-        )
-
-        @JvmStatic
-        @BeforeAll
-        fun setupRetryPolicyTest() {
-            setup(appFactoryForEc1 = { consulPort -> EnvoyControlRunnerTestApp(properties, consulPort) })
-        }
+        @JvmField
```

Comment:
Do extensions have to be `@JvmField` or is there some other reason?

Reply:
Kotlin-JUnit incompatibility peculiarities. In short, you need `@JvmField` together with `@RegisterExtension` in Kotlin.